### PR TITLE
Notify the user of new changes in the active editable

### LINKED ISSUE #4583: Notify the user if a new action has been done in the active editable
It would be a very interesting feature to be able to get live notifications when an editable someone is currently visiting no longer has the most up to date state.
Imagine the following use-case:
- A user visits an editable (the page will remain open)
- An editor immediately makes changes
- The user sees a notification telling him that the editable has new changes

Do not confuse the term _notification_ with system-wide notifications. The purpose of this issue is to implement this feature locally, such as polling for editable changes every X seconds.

This would be particularly helpful in #4572, as replaced reviews (such as watermarking) are placed shortly after the editable has been submitted, resulting in an immediate outdated state.

### PR DESCRIPTION
**Pull request by plourenco**
Closes #4583 

Implements simple polling that compares with the previous response data (before going into the reducer).
Will display a message such as:
image

The interval is flexible, currently defaults to 10 seconds.

### CODE REVIEW

File: indico/modules/events/editing/client/js/editing/timeline/index.jsx [hunk 1]
```diff
@@ -5,19 +5,23 @@
...
 import TimelineContent from 'indico/modules/events/reviewing/components/TimelineContent';
+import {indicoAxios} from 'indico/utils/axios';
 import SubmitRevision from './SubmitRevision';
 
 import * as actions from './actions';
 import * as selectors from './selectors';
 import TimelineItem from './TimelineItem';
 import FileDisplay from './FileDisplay';
 
+const POOLING_SECONDS = 10;
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/editing/timeline/index.jsx:23]: `POLLING_SECONDS`

File: indico/modules/events/editing/client/js/editing/timeline/index.jsx [hunk 2]
```diff
@@ -40,6 +71,17 @@ export default function Timeline() {
 
   return (
     <>
+      {isOutdated && (
+        <Message
+          warning
+          header="This revision has been updated"
+          content={
+            <span>
+              <a onClick={refresh}>Click here to refresh</a> and see the most recent changes.
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/editing/timeline/index.jsx:80]: i18n

Comment (ThiefMaster) [reply indico/modules/events/editing/client/js/editing/timeline/index.jsx:80]: most recent *version*?

File: indico/modules/events/editing/client/js/editing/timeline/index.jsx [hunk 3]
```diff
@@ -5,19 +5,23 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
-import React, {useEffect} from 'react';
+import React, {useEffect, useRef, useState} from 'react';
 import {useDispatch, useSelector} from 'react-redux';
-import {Loader} from 'semantic-ui-react';
+import {Loader, Message} from 'semantic-ui-react';
 
+import _ from 'lodash';
```
Comment (ThiefMaster) [inline indico/modules/events/editing/client/js/editing/timeline/index.jsx:12]: move this one up to the previous block of imports

File: indico/modules/events/editing/client/js/editing/timeline/index.jsx (hunk 1 above)
Comment (plourenco) [reply indico/modules/events/editing/client/js/editing/timeline/index.jsx]: 😪

File: indico/modules/events/editing/client/js/editing/timeline/index.jsx [hunk 4]
```diff
@@ -40,6 +72,20 @@ export default function Timeline() {
 
   return (
     <>
+      {isOutdated && (
+        <Message
+          warning
+          header="This revision has been updated"
```
Comment (panagiotappl) [inline indico/modules/events/editing/client/js/editing/timeline/index.jsx:78]: I think you could add translation here

Comment (plourenco) [reply indico/modules/events/editing/client/js/editing/timeline/index.jsx:78]: Thanks!

**Pull request merged**